runtime/onert/backend/cpu/KernelGenerator.cc

```diff
@@ -921,7 +921,6 @@ void KernelGenerator::visit(const ir::operation::Split &node)
   const auto rank = node.param().rank;
   const auto axis = ops::getAxis(rank, node.param().axis, _current_op_seq_layout);
   auto axis_resolved = axis < 0 ? axis + rank : axis;
-  assert(0 <= axis_resolved && axis_resolved < rank);
```

Comment:
Is this relevant to this PR ?

Reply:
Yes, when enter a dynamic tensor as input, the rank will be 0.
So, I move that condition into shape inf layer.

Reply:
KernelGenerator::visit(const ir::operation::Split &node) called when compile time. So if input is a dynamic tensor, the it  couldn't get a intact input's shape. Even though,  "assert(0 <= axis_resolved && axis_resolved < rank)" tries to check rank of input. In this situation, It will ouccurr an assertion always.